service/statistics/AnalyticsEvents.js

```diff
@@ -1,447 +1,414 @@
+ * events a factory function is easier.
  *
- * The naming convention for the constants below uses "_" as a prefix or
- * suffix to indicate that known usage of the constant prepends or appends
- * a string to the name of the event.
- */
-
-/**
- * Properties: value
+ * A general approach for adding a new event:
+ * 1. Determine the event type: track, UI, page, or operational. If in doubt use
+ * operational.
+ * 2. Determine whether the event is related to other existing events, and
+ * which fields are desired to be set: name, action, actionSubject, source.
+ * 3. If the name is sufficient (the other fields are not important), use a
+ * constant. Otherwise use a factory function.
  *
- * TODO: document, reformat
+ * Note that the AnalyticsAdapter uses the events passed to its functions for
+ * its own purposes, and might modify them. Because of this factory functions
+ * should create new objects.
  *
- * Full event names (uncertain):
- * conference.muc.joined (???)
- * conference.sharingDesktop.start (???)
- * conference.sharingDesktop.stop (???)
- * xmpp.attached (???)
- * xmpp.attaching (???)
- * xmpp.connected (???)
- * xmpp.connecting (???)
- * xmpp.session-initiate (???)
  */
-export const _CONNECTION_TIMES_ = '';
 
 /**
- * TODO: document, reformat (group together with other ICE events)
- *
- * Known full event names:
- * ice.initiator.checksDuration
- * ice.responder.checksDuration
- * p2p.ice.initiator.checksDuration
- * p2p.ice.responder.checksDuration
- *
- * Properties: value
+ * The constant which identifies an event of type "operational".
+ * @type {string}
  */
-export const _ICE_CHECKING_DURATION = 'checksDuration';
+export const TYPE_OPERATIONAL = 'operational';
 
 /**
- * TODO: document, reformat
- *
- * Known full event names:
- * ice.checking
- * ice.closed
- * ice.completed
- * ice.connected
- * ice.disconnected
- * ice.failed
- * p2p.ice.checking
- * p2p.ice.closed
- * p2p.ice.completed
- * p2p.ice.connected
- * p2p.ice.disconnected
- * p2p.ice.failed
- *
- * Properties: value
+ * The constant which identifies an event of type "page".
+ * @type {string}
  */
-export const _ICE_CONNECTION_STATE_ = 'ice';
+export const TYPE_PAGE = 'page';
 
 /**
- * TODO: document, reformat (group together with other ICE events)
- *
- * Known full event names:
- * ice.initiator.establishmentDuration
- * ice.responder.establishmentDuration
- * p2p.ice.initiator.establishmentDuration
- * p2p.ice.responder.establishmentDuration
- *
- * Properties: value
+ * The constant which identifies an event of type "track".
+ * @type {string}
  */
-export const _ICE_ESTABLISHMENT_DURATION = 'establishmentDuration';
+export const TYPE_TRACK = 'track';
 
 /**
- * TODO: document, reformat
- *
- * Known full event names:
- * ice.initiator.gatheringDuration
- * ice.responder.gatheringDuration
- * p2p.ice.initiator.gatheringDuration
- * p2p.ice.responder.gatheringDuration
- *
- * Properties: value
+ * The constant which identifies an event of type "ui".
```

Comment:
Maybe these type comments could briefly explain what each type is for.

Reply:
I'm still trying to figure that out :(

Reply:
Okay, yeah, I don't understand the difference either fully either...